okhttp/src/main/java/okhttp3/internal/platform/ConscryptPlatform.java

```diff
@@ -0,0 +1,99 @@
+package okhttp3.internal.platform;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.Provider;
+import java.util.List;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.X509TrustManager;
+import okhttp3.Protocol;
+import org.conscrypt.Conscrypt;
+import org.conscrypt.OpenSSLProvider;
+
+/**
+ * Platform using Conscrypt (conscrypt.org) if system property
+ * okhttp.platform is set to conscrypt.
+ *
+ * Requires org.conscrypt:conscrypt-openjdk-uber on the classpath.
+ */
+public class ConscryptPlatform extends Platform {
+  private ConscryptPlatform() {
+  }
+
+  public Provider getProvider() {
+    return new OpenSSLProvider();
+  }
+
+  @Override public X509TrustManager trustManager(SSLSocketFactory sslSocketFactory) {
+    if (Conscrypt.isConscrypt(sslSocketFactory)) {
```

Comment:
simpler to invert this and remove a layer of indentation?

```
if (!Conscrypt.isConscrypt(sslSocketFactory)) {
  return super.trustManager(sslSocketFactory);
}
```

?

Reply:
ALSO – should we fail if we’re using a Concrypt platform but not its SSL sockets? That’s a weird situation to be in and it might be more friendly to just explode?